pkgs/desktops/cinnamon/cinnamon-common/default.nix

```diff
@@ -156,6 +148,14 @@ stdenv.mkDerivation rec {
     sed "s|/usr/bin|$out/bin|g" -i ./files/usr/share/xsessions/cinnamon.desktop ./files/usr/share/xsessions/cinnamon2d.desktop
   '';
 
+  preFixup = '' # it looks for the lib in the wrong place
+    pushd $out/lib
+    for f in cinnamon/*; do
```

Comment:
not sure if we could change something in the meson file instead - or what even is causing this difference between 4.6

Reply:
The thing trying to load libraries from needs `libdir / 'cinnamon/foo'` will need that in `install_rpath`. Example https://gitlab.gnome.org/GNOME/gnome-documents/-/commit/d18a92e0a940073ac766f609937539e4fc6cdbb7

Reply:
Perhaps https://github.com/linuxmint/cinnamon/commit/1b52e96ecf499f1a7d7c8c71c1aff39d88dca04d but that is supposed to be already merged.

Reply:
Yeah @mkg20001, seen this issue a couple times so it would be worth a patch to try.